hushline/premium.py

```diff
@@ -0,0 +1,609 @@
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=True)
+                user.stripe_subscription_cancel_at_period_end = True
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Autorenew disabled for subscription {user.stripe_subscription_id} for user {user.id}"  # noqa: E501
+                )
+
+                flash("Autorenew has been disabled.")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}")
+                return jsonify(success=False), 400
+
+        return jsonify(success=False), 400
+
+    @bp.route("/enable-autorenew", methods=["POST"])
+    @authentication_required
+    def enable_autorenew() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=False)
+                user.stripe_subscription_cancel_at_period_end = False
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Autorenew enabled for subscription {user.stripe_subscription_id} for user {user.id}"  # noqa: E501
+                )
+
+                flash("Autorenew has been enabled.")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}")
+                return jsonify(success=False), 400
+
+        return jsonify(success=False), 400
+
+    @bp.route("/cancel", methods=["POST"])
+    @authentication_required
+    def cancel() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                # Cancel the subscription
+                stripe.Subscription.delete(user.stripe_subscription_id)
+
+                # Downgrade the user (the subscription ID will get removed in the webhook)
+                user.tier_id = FREE_TIER
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Subscription {user.stripe_subscription_id} canceled for user {user.id}"
+                )
+
+                flash("💔 Sorry to see you go!")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
```

Comment:
Might want `exc_info=True` in the logger to get more details instead of `{e}`

Reply:
Fixed in 977935a15167450c085c93340cb31e0ac6efca01